compiler/backend/src/org/jetbrains/kotlin/codegen/coroutines/CoroutineTransformerMethodVisitor.kt

```diff
@@ -702,11 +703,12 @@ class CoroutineTransformerMethodVisitor(
         id: Int,
         suspension: SuspensionPoint,
         methodNode: MethodNode,
-        suspendMarkerVarIndex: Int
+        suspendMarkerVarIndex: Int,
+        suspendPointLineNumber: LineNumberNode?
     ): LabelNode {
         val continuationLabel = LabelNode()
         val continuationLabelAfterLoadedResult = LabelNode()
-        val suspendElementLineNumber = lineNumber
+        val suspendElementLineNumber = suspendPointLineNumber?.line ?: lineNumber
         var nextLineNumberNode = suspension.suspensionCallEnd.findNextOrNull { it is LineNumberNode } as? LineNumberNode
```

Comment:
This looks shaky to me. Scanning forward ignores control flow. Also, I would guess that for code on one line you might end up having one branch look like it is on the next line. So I think there is a bug here where you can get the wrong line number when you resume. I'll see if I can generate a test case and attempt a fix.

Reply:
There is a bug. In fact, there are plenty of them. Every time a suspending call is placed at the same line with other suspending calls/throw expression, there will be a bug.
Some of them are fixed `suspendMe() ?: error("")`, some are not `suspendMe() + suspendMe()` etc.

Reply:
Yes, after writing this comment I created an example that just had two throws on the same line:

```
if (b) throw A() else throw A()
```

and one of them indeed gets the wrong line number.